.circleci/config.yml

```diff
@@ -0,0 +1,113 @@
+version: 2.1
+
+executors:
+  my-executor:
+    docker:
+      - image: continuumio/miniconda3
+      # astroconda/python:3.6.1
+    working_directory: ~/repo
+
+jobs:
+  build:
+    executor: my-executor
+    environment:
+      - SOURCE_BRANCH: master
+      - TARGET_BRANCH: gh-pages
+    steps:
+      - add_ssh_keys
+
+      - checkout
+
+      - restore_cache:
+          key: v1-miniconda-{{ .Branch }}
+
+      - run:
+          name: Install miniconda
+          command: |
+            ./.circleci/install_conda.sh
+
+      - save_cache:
+          key: v1-miniconda-{{ .Branch }}
+          paths:
+            - /home/circleci/miniconda
+
+      - restore_cache:
+          key: v1-miniconda-depends-{{ .Branch }}
+
+      - run:
+          name: Setup Environment
+          command: |
+            ./.circleci/setup_env.sh
+
+      - save_cache:
+          key: v1-miniconda-depends-{{ .Branch }}
+          paths:
+            - /home/circleci/miniconda
+
+      - run:
+          name: Run/convert notebooks
+          command: |
+            source activate /home/circleci/miniconda/envs/notebooks_env
+            conda info --envs
+            python ./convert.py
+
+      - run:
+          name: Check notebooks
+          command: |
+            COMMIT_RANGE=$(echo "${CIRCLE_COMPARE_URL}" | cut -d/ -f7)
+            if [[ $COMMIT_RANGE != *"..."* ]]; then
+              COMMIT_RANGE="${COMMIT_RANGE}...${COMMIT_RANGE}"
+            fi
+            source activate /home/circleci/miniconda/envs/notebooks_env
+            conda info --envs
+            python -m "nbpages.check_nbs" --commit-range=${COMMIT_RANGE}
+
+      # clone gh-pages branch, copy over files from build, commit and push
+      - deploy:
+          name: Deploy
+          command: |
+            git config --global user.email $GH_EMAIL
+            git config --global user.name $GH_NAME
+
+            git clone $CIRCLE_REPOSITORY_URL /out
```

Comment:
From here on (in the "deploy" stage) it would be good for this to be a separate file that can be manually invoked.  That way it can be shared with other CI implementations and if need be the build/deploy can be entirely manually done if we run into CI problems but really need to get a notebook out or something.